src/Sarif.Multitool/Rules/SARIF2005.ProvideHelpfulToolInformation.cs

```diff
@@ -0,0 +1,93 @@
+    {
+        public override MultiformatMessageString FullDescription => new MultiformatMessageString
+        {
+            Text = RuleResources.SARIF2005_ProvideHelpfulToolInformation_FullDescription_Text
+        };
+
+        public override FailureLevel DefaultLevel => FailureLevel.Warning;
+
+        public override string Id => RuleId.ProvideHelpfulToolInformation;
+
+        protected override IEnumerable<string> MessageResourceNames => new string[]
+        {
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_ProvideToolVersion_Text),
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_ProvideConciseToolName_Text),
+            nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_UseNumericToolVersions_Text)
+        };
+
+        protected override void Analyze(Tool tool, string toolPointer)
+        {
+            if (tool == null)
+            {
+                return;
+            }
+
+            if (tool.Driver != null)
+            {
+                AnalyzeToolDriver(tool.Driver, toolPointer);
+            }
+        }
+
+        private void AnalyzeToolDriver(ToolComponent toolComponent, string toolPointer)
+        {
+            // ProvideConciseToolName: Ensure that tool.driver.name isn't more than 3 words long
+            if (!string.IsNullOrEmpty(toolComponent.Name))
+            {
+                const int maxWordCount = 3;
+                int wordCount = toolComponent.Name.Split(' ').Length;
+                if (wordCount > maxWordCount)
+                {
+                    string driverPointer = toolPointer.AtProperty("driver").AtProperty("name");
+                    LogResult(
+                        driverPointer,
+                        nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_ProvideConciseToolName_Text),
+                        wordCount.ToString(),
+                        maxWordCount.ToString());
+                }
+            }
+
+            // ProvideToolVersion: Either tool.driver.version or tool.driver.semanticVersion should be there.
+            if (string.IsNullOrWhiteSpace(toolComponent.Version) && string.IsNullOrWhiteSpace(toolComponent.SemanticVersion))
+            {
+                LogResult(
+                        toolPointer,
+                        nameof(RuleResources.SARIF2005_ProvideHelpfulToolInformation_Warning_ProvideToolVersion_Text));
+            }
+            else
+            {
+                // UseNumericToolVersions
+                if (!string.IsNullOrWhiteSpace(toolComponent.Version))
+                {
+                    AnalyzeVersion(toolComponent.Version, toolPointer.AtProperty("driver").AtProperty("version"));
+                }
+                else
+                {
+                    AnalyzeVersion(toolComponent.SemanticVersion, toolPointer.AtProperty("driver").AtProperty("semanticVersion"));
```

Comment:
>AnalyzeVersion(toolComponent.SemanticVersion, toolPointer.AtProperty("driver").AtProperty("semanticVersion")); [](start = 20, length = 110)

The `semanticVersion` doesn't match the simple regex we are using for `version`. It actually matches the complicated expression you showed me earlier. Hang on while I find the official source for this...

Reply:
See the regexes at the end of SemVer.org.

---
In reply to: [444576559](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444576559) [](ancestors = 444576559)

Reply:
ok. just created another static regex and passed as variable in the method

---
In reply to: [444577239](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444577239) [](ancestors = 444577239,444576559)

Reply:
The problem with that is that the message will be wrong. For `semanticVersion` you'd want to say something like "semanticVersion doesn't match the SemVer.org specification". For `version` you'd want to say "Version isn't numeric".

---
In reply to: [444578598](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444578598) [](ancestors = 444578598,444577239,444576559)

Reply:
For now, _do not verify_ `semanticVersion`. Just remove this branch of the logic. There needs to be a separate rule for `semanticVersion`.

---
In reply to: [444579925](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444579925) [](ancestors = 444579925,444578598,444577239,444576559)